### PR TITLE
Feature/#12 system envs

### PR DESCRIPTION
**Pull request by dmitry-osin**
Hello, @chaos-adept 
Could you please take a look my PR, below all main changes that I've made in this branch:

1. Implement python script that allows to set/unset environment variables depends of params count
2. Use this script to set/unset environment variables during apply/destroy stages of terraform
3. Implement bash script that allows to generate local_specific_variables.h based on environment variables which were set by terraform on apply stage
4. Refactor publish/send scripts to use environment variables that were set by terraform on apply stage

### CODE REVIEW

File: tests/manual/iot-core/publish-from-registry-to-device-commands.sh [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
-#REGISTRY_PASSWORD="zx+z{i1QD6YJ#J#&"
-#DEVICE_ID="arekifdlttq12q6sslle"
-#DEVICE_PASSWORD="HsTvGAN6HFundl1v"
+REGISTRY_ID=$(echo "$IOT_REGISTRY")
+REGISTRY_PASSWORD=$(echo "$IOT_REGISTRY_PASSWORD")
 
+EMPTY_LINE=""
+
+echo "--------------------------------------------------------"
+echo "Credentials: '${REGISTRY_ID}' : '${REGISTRY_PASSWORD}'"
```
Comment (chaos-adept) [inline tests/manual/iot-core/publish-from-registry-to-device-commands.sh:9]: @dmitry-osin 
does it make sense to hide the password? 

might we have the following case?
if we forget to remove it and this script is used in the github actions then we might compromise the test cloud folder registry.

as result, a bad person can just spam mqtt messages and increase cloud costs

File: tests/manual/iot-core/subscribe-device-to-registry-commands.sh [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
-#REGISTRY_PASSWORD="zx+z{i1QD6YJ#J#&"
-#DEVICE_ID="arekifdlttq12q6sslle"
-#DEVICE_PASSWORD="HsTvGAN6HFundl1v"
+REGISTRY_ID=$(echo "$IOT_REGISTRY")
+REGISTRY_PASSWORD=$(echo "$IOT_REGISTRY_PASSWORD")
 
+EMPTY_LINE=""
+
+echo "--------------------------------------------------------"
+echo "Credentials: '${REGISTRY_ID}' : '${REGISTRY_PASSWORD}'"
```
Comment (chaos-adept) [inline tests/manual/iot-core/subscribe-device-to-registry-commands.sh:9]: the same as [here](https://github.com/chaos-adept/iot-dipstick/pull/30/files#diff-d409215c15980a38f6fa6df07f6d55a15e98d30a0c8718f7e5690f43bbe1e926R9)

**Review by chaos-adept — COMMENTED** [review]
`helpers` is a folder for scripts, it is a quite unusual name, is it common practice for python? might the name be `scripts`?

**Review by chaos-adept — COMMENTED** [review]
@dmitry-osin 
is it possible to move all env generation related terraform code in the one place in the separated terraform "module"/"script", something like `inject_local_env_vars.tf` or just in the `output.tf`?

**Review by chaos-adept — CHANGES_REQUESTED** [review]
also there is an issue with destroy 
Could it be fixed?
image

File: tests/manual/iot-core/publish-from-registry-to-device-commands.sh (hunk 1 above)
Comment (dmitry-osin) [reply tests/manual/iot-core/publish-from-registry-to-device-commands.sh:9]: Thanks for reply. I believe that these values will be taken from dev machine envs and you do not need to modify this script.

File: tests/manual/iot-core/subscribe-device-to-registry-commands.sh (hunk 2 above)
Comment (dmitry-osin) [reply tests/manual/iot-core/subscribe-device-to-registry-commands.sh:9]: Same here too :D

**Comment by dmitry-osin** [comment]
> also there is an issue with destroy
> Could it be fixed?
> image

As we discussed, could you please add python to path environment variable?

**Comment by dmitry-osin** [comment]
> @dmitry-osin
> is it possible to move all env generation related terraform code in the one place in the separated terraform "module"/"script", something like `inject_local_env_vars.tf` or just in the `output.tf`?

good stuff, I'll try it

File: terraform/inject_local_env_vars.tf [hunk 3]
```diff
@@ -0,0 +1,94 @@
+resource "null_resource" "iot_registry" {
```
Comment (dmitry-osin) [inline terraform/inject_local_env_vars.tf:1]: @chaos-adept could you please take a look my PR if you have spare a time?

**Comment by chaos-adept** [comment]
changes are fine now, thanks!
it is useful

**Pull request merged**